R/OptimInstance.R

```diff
@@ -120,6 +120,41 @@ OptimInstance = R6Class("OptimInstance",
       assert_names(names(y), permutation.of = self$objective$codomain$ids())
       x_domain = transform_xdt_to_xss(xdt, self$search_space)[[1]]
       private$.result = cbind(xdt, x_domain = list(x_domain), t(y)) # t(y) so the name of y stays
+    },
+
+    #' @description
+    #' Evaluates (untransformed) points of only numeric values, and returns a
+    #' scalar objective value, where the return value is negated if the measure
+    #' is maximized. Internally, `$eval_batch()` is called with a single row.
+    #' This function serves as a objective function for optimizers of numeric
+    #' spaces - which should always be minimized.
+    #'
+    #' @param x (`numeric()`)\cr
+    #' Untransformed points.
+    #'
+    #' @return Objective value as `numeric(1)`.
+    objective_function = function(x) {
```

Comment:
We should also assert that the `search_space` is purely numeric and return a good error message in case it is not. e.g. "`$objective_function` can only be called if `search_space` only contains numeric values". Actually it would be probably a good design to set the objective function upon creation of the `OptimInstance`. So `objective_function(x)` would call the private `.objective_function(self, x)`. This private function can point to various functions such as `objective_function_numeric_min_min(instance, x)`, `objective_numeric_max_min(instance, x)` or simply point to a function that creates error when the `search_space` in not entirely numeric. But maybe it also makes things too complicated.

Reply:
We could use a simple assert for now and transfer your comment to an issue since the api stays the same? It would be nice to have this merged, so we can work on https://github.com/mlr-org/mlr3tuning/pull/248

Reply:
> We could use a simple assert for now and transfer your comment to an issue since the api stays the same? It would be nice to have this merged, so we can work on [mlr-org/mlr3tuning#248](https://github.com/mlr-org/mlr3tuning/pull/248)

pls do that